### PR TITLE
Remove impl_outer_ macros

### PR DESCRIPTION
**Pull request by thiolliere**
Those macro are no longer used, people should rather use construct_runtime, it is some code debt when implementing new functionality in substrate (e.g. metadata changes).

### CODE REVIEW

File: frame/support/src/dispatch.rs [hunk 1]
```diff
@@ -2901,26 +2785,12 @@ mod tests {
...
 	fn get_call_names() {
 		let call_names = Call::<TraitImpl>::get_call_names();
 		assert_eq!(["aux_0", "aux_1", "aux_2", "aux_3", "aux_4", "aux_5", "operational"], call_names);
 	}
 
-	#[test]
-	fn get_module_names() {
-		let module_names = OuterCall::get_module_names();
-		assert_eq!(["Test"], module_names);
-	}
```
Comment (thiolliere) [inline frame/support/src/dispatch.rs:2922]: those test are in frame/support/test/tests/construct_runtime.rs

File: frame/support/src/lib.rs [hunk 2]
```diff
@@ -49,26 +49,29 @@ pub use log;
...
 pub mod instances;
 
+#[doc(hidden)]
+pub mod unsigned {
+	#[doc(hidden)]
+	pub use crate::sp_runtime::traits::ValidateUnsigned;
+	#[doc(hidden)]
+	pub use crate::sp_runtime::transaction_validity::{
+		TransactionValidity, UnknownTransaction, TransactionValidityError, TransactionSource,
+	};
```
Comment (thiolliere) [inline frame/support/src/lib.rs:72]: all those reexport were already hidden, some are used by construct_runtime, some are used by some pallet.

**Review by shawntabrizi — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Comment by thiolliere** [comment]
farewell macro rules

**Comment by thiolliere** [comment]
bot merge

**Pull request merged**